Makefile

```diff
@@ -0,0 +1,66 @@
+PROJECT := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+
+SRC := $(PROJECT)src
+TESTS := $(PROJECT)tests
+ALL := $(SRC) $(TESTS)
+
+export PYTHONPATH = $(PROJECT):$(PROJECT)/lib:$(SRC)
+export PY_COLORS=1
+
+lock:
+	uv lock --upgrade --no-cache
+
+clean-requirements:
+	rm -f requirements*.txt
+
+generate-requirements: clean-requirements
+	uv pip compile -q --no-cache pyproject.toml -o requirements.txt
+
+lint:
+	uv tool run ruff check $(ALL)
+	uv tool run ruff format --check --diff $(ALL)
+
+static:
+	uv run --extra static pyright
+
+fmt:
+	uv tool run ruff check --fix-only $(ALL)
+	uv tool run ruff format $(ALL)
+
+unit:
+	uv run --isolated \
+	    --extra unit \
+		coverage run \
+		--source=$(SRC) \
+		-m pytest \
+		--ignore=$(TESTS)/integration \
```

Comment:
`unit` tests everything in this repo, but then uses `--ignore` to omit the integration tests.  scenario and integration test only specific folders.  We should probably use just one of those two methods

also, I think `unit` test right now will run sidecar tests too

Reply:
Maybe there is a typo in your comment or I do not fully understand, do you mean to remove the --ignore because it is the same as this conversation:

https://github.com/canonical/tempo-coordinator-k8s-operator/pull/62#discussion_r1842576307